velox/common/process/TraceContext.cpp

```diff
@@ -41,7 +42,16 @@ TraceContext::TraceContext(std::string label, bool isTemporary)
     entry.time = enterTime_;
     entry.file = __FILE__;
     entry.line = __LINE__;
-    snprintf(entry.label, entry.kLabelCapacity, "%s", label_.c_str());
+    int result = snprintf(entry.label, entry.kLabelCapacity, "%s", label_.c_str());
+
+    if (result < 0) {
+      throw std::runtime_error("Encoding error in snprintf");
```

Comment:
On error I suggest to set the first byte to null byte. That's all that is needed to ensure the buffer doesn't contains anything bad. I think the buffer itself is not initialized prior so could contain random data which is overwritten or on error we set the null byte at the beginning. This should take care of errors and use of an uninitialized buffer.

Reply:
Yes we can do that.  This virtually never happens though so we should wrap the condition inside `FOLLY_UNLIKELY` so CPU don't waste time decode the unused instructions.

Reply:
Done.